### PR TITLE
[TASK] Document our approach to assertions and type checks

### LINKED ISSUE #1330: Document checking type and value of variables: assertions, exceptions
Document why we should use assert() instead of dedicated libraries

### PR DESCRIPTION
**Pull request by oliverklee**
Fixes #1330

### CODE REVIEW

**Comment by lukaszuznanski** [comment]
Should this way for decision for ADR/Spikes and then document it this way or documentation is just fine?

**Comment by oliverklee** [comment]
In our [last call](https://git.typo3.org/qa/internal-organization/-/blob/master/MeetingMinutes/2024/05/27.md?ref_type=heads), we decided this:

> Oli already has experience with that and will add a PR to EXT:tea documentation to document the best practices, so we all can learn and follow the advice.

So we can use this as-is without requiring an additional decision.

File: Documentation/Assertions.rst [hunk 1]
```diff
@@ -0,0 +1,72 @@
+.. include:: /Includes.rst.txt
+
+.. _assertions-and-type-checks:
+
+==========================================
+Our approach to assertions and type checks
+==========================================
```
Comment (DanielSiepmann) [inline Documentation/Assertions.rst:7]:
```suggestion
=========================================
Our approach to assertions and type checks
=========================================
```

**Review by DanielSiepmann — APPROVED** [review]
Thanks for taking the time and solving the issue.

I especially like that you included concrete code examples which makes it easier to grasp and understand as a developer.

Just one small minor RST thing which won't break, that's why I approved already.

File: Documentation/Assertions.rst (hunk 1 above)
Comment (oliverklee) [reply Documentation/Assertions.rst:7]: I'm confused … on my screen, your change makes the `===` lines 1 char shorter than the text line, and AFAIK the `===` lines should be the same length. Am I missing something here?

Comment (DanielSiepmann) [reply Documentation/Assertions.rst:7]: They should. Looks like an issue with GitHub on my end than.
https://raw.githubusercontent.com/FriendsOfTYPO3/tea/3949ed3ea7cfc4ed58ebfb94a2be1c41889e97a4/Documentation/Assertions.rst looks okay.

Screenshot for reference:
image

**Pull request merged**